cmd/helm/installer/init_test.go

```diff
@@ -97,6 +99,35 @@ func TestEnsureHome(t *testing.T) {
 		t.Error(err)
 	}
 
+	rf, err := repo.LoadRepositoriesFile(hh.RepositoryFile())
+	if err != nil {
+		t.Error(err)
+	}
+
+	foundStable := false
+	stableCachePath := fmt.Sprintf("%s-index.yaml", stableRepository)
+	foundLocal := false
+	localCachePath := fmt.Sprintf("%s-index.yaml", LocalRepository)
+	for _, rr := range rf.Repositories {
```

Comment:
So this test is still checking just the cache location instead of the fully generated cache name. I'd recommend changing these tests to do something like this to make sure the full path it generates using the relative paths is correct:

```diff
diff --git a/cmd/helm/installer/init_test.go b/cmd/helm/installer/init_test.go
index 40c83589..4e0a0d38 100644
--- a/cmd/helm/installer/init_test.go
+++ b/cmd/helm/installer/init_test.go
@@ -21,6 +21,7 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"path"
 	"testing"
 
 	helm_env "k8s.io/helm/pkg/helm/environment"
@@ -104,29 +105,21 @@ func TestEnsureHome(t *testing.T) {
 		t.Error(err)
 	}
 
-	foundStable := false
 	stableCachePath := fmt.Sprintf("%s-index.yaml", stableRepository)
-	foundLocal := false
 	localCachePath := fmt.Sprintf("%s-index.yaml", LocalRepository)
 	for _, rr := range rf.Repositories {
 		if rr.Name == stableRepository {
-			if rr.Cache != stableCachePath {
-				t.Errorf("stable repo cache path is %s, not %s", rr.Cache, stableCachePath)
+			expected := path.Join(home, "repository", "cache", stableCachePath)
+			if hh.CacheIndex(stableRepository) != expected {
+				t.Errorf("Incorrect stable cache path.\nExpected: %s\nGot: %s", expected, hh.CacheIndex(stableRepository))
 			}
-			foundStable = true
 		} else if rr.Name == LocalRepository {
-			if rr.Cache != localCachePath {
-				t.Errorf("local repo cache path is %s, not %s", rr.Cache, localCachePath)
+			expected := path.Join(home, "repository", "cache", localCachePath)
+			if hh.CacheIndex(LocalRepository) != expected {
+				t.Errorf("Incorrect stable cache path.\nExpected: %s\nGot: %s", expected, hh.CacheIndex(LocalRepository))
 			}
-			foundLocal = true
 		}
 	}
-	if !foundStable {
-		t.Errorf("stable repo not found")
-	}
-	if !foundLocal {
-		t.Errorf("local repo not found")
-	}
 
 	expectedDirs := []string{hh.String(), hh.Repository(), hh.Cache(), hh.LocalRepository()}
 	for _, dir := range expectedDirs {
```